File: frontend/src/services/entrenamiento.service.ts
import { Injectable, inject } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { API_BASE_URL } from "./api.config";
import { Entrenamiento } from "../models/entrenamiento.model";

@Injectable({
  providedIn: "root",
})
export class EntrenamientoService {
  private http = inject(HttpClient);
  private apiUrl = `${API_BASE_URL}/entrenamientos`;

  crear(entrenamiento: Entrenamiento): Observable<Entrenamiento> {
    return this.http.post<Entrenamiento>(this.apiUrl, entrenamiento);
  }

  listar(): Observable<Entrenamiento[]> {
    return this.http.get<Entrenamiento[]>(this.apiUrl);
  }

  obtenerPorId(id: number): Observable<Entrenamiento> {
    return this.http.get<Entrenamiento>(`${this.apiUrl}/${id}`);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: backend_entrenemos/src/main/java/com/entrenemos/services/EntrenamientoService.java
package com.entrenemos.services;

import com.entrenemos.dto.EntrenamientoDto;

import java.util.List;

public interface EntrenamientoService {
    EntrenamientoDto crear(EntrenamientoDto entrenamientoDto);
    EntrenamientoDto actualizar(Long id, EntrenamientoDto entrenamientoDto);
    List<EntrenamientoDto> listar();
    EntrenamientoDto obtenerPorId(Long id);
    void eliminar(Long id);
}
